packages/myst-templates/src/validators.ts

```diff
@@ -312,11 +315,42 @@ export function validateTemplateOptionDefinition(
     });
   }
   if (defined(value.max_chars)) {
-    output.max_chars = validateNumber(value.max_chars, {
-      min: 0,
-      integer: true,
-      ...incrementOptions('max_chars', opts),
-    });
+    if (output.type === 'string') {
+      output.max_chars = validateNumber(value.max_chars, {
+        min: 0,
+        integer: true,
+        ...incrementOptions('max_chars', opts),
+      });
+    } else {
+      validationError('type must be "string" to use "max_chars" option', opts);
```

Comment:
@fwkoch this is true correct?

Reply:
Maybe we add a similar constraint for `choices` only on `type: 'choice'`? (We could also just drop `type: 'choice'` in favour of `type: 'string'` with `choices` defined...)

Reply:
Added.